### PR TITLE
backend: Filter web search domains for Tavily

### PR DESCRIPTION
**Pull request by tianjing-li**
- Adds domain restriction for Tavily web search tool
- TODO: add for google/brave when they are implemented

### CODE REVIEW

**Review by raphael-cohere — APPROVED** [review]

File: src/backend/crud/agent_tool_metadata.py [hunk 1]
```diff
@@ -61,6 +61,35 @@ def get_all_agent_tool_metadata_by_agent_id(
     )
 
 
+def get_agent_tool_metadata(
+    db: Session,
+    agent_id: str,
+    tool_name: str,
+    user_id: str,
+) -> list[AgentToolMetadata]:
```
[the remark this answers is not in the document]
Comment (tianjing-li) [reply src/backend/crud/agent_tool_metadata.py:69]: done

File: src/backend/tools/tavily.py [hunk 2]
```diff
@@ -4,28 +4,60 @@
...
 
     @classmethod
     def is_available(cls) -> bool:
         return cls.TAVILY_API_KEY is not None
 
+    def get_filtered_domains(self, session: DBSessionDep, ctx: Any):
+        agent_id = ctx.get_agent_id()
+        user_id = ctx.get_user_id()
+
+        if agent_id is None or user_id is None:
```
[the remark this answers is not in the document]
Comment (tianjing-li) [reply src/backend/tools/tavily.py:29]: yep that works for me.

File: src/backend/tools/tavily.py [hunk 3]
```diff
@@ -4,28 +4,60 @@
...
 
     def __init__(self):
         self.client = TavilyClient(api_key=self.TAVILY_API_KEY)
-        self.num_results = 6
 
     @classmethod
     def is_available(cls) -> bool:
         return cls.TAVILY_API_KEY is not None
 
+    def get_filtered_domains(self, session: DBSessionDep, ctx: Any):
```
[the remark this answers is not in the document]
Comment (tianjing-li) [reply src/backend/tools/tavily.py:25]: Works for me too. Will have to slightly change call method handling

[the remark this answers is not in the document]
Comment (tianjing-li) [reply src/backend/tools/tavily.py]: Done. Had to add handling in the `call` method since `include_domains` should be None of no filtering is to be done.

**Pull request merged**